src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java

```diff
@@ -110,8 +113,12 @@ public Payment createPayment(
     }
 
     public void cancelPayment(String authorisation, String paymentReference) {
-        logger.info("Cancelling payment {}", paymentReference);
-        paymentsClient.cancelPayment(authorisation, paymentReference);
+        if (autoCancel) {
```

Comment:
My preference would be if this flag is off don't even call the method `cancelPayment`.

Reply:
Agreed. See https://github.com/hmcts/cmc-claim-store/pull/1587/commits/a6055e9948254eccdabd140bed6e40e9d8892320